# Homework 2 - Data Wrangling with pandas

## Preparation

In [1]:
# Importing libraries
import pandas as pd
import numpy as np
import os
os.getcwd()

os.chdir(r"C:/Users/natal/OneDrive/Documents/GitHub/QMSS-GR5072-Spring2026-HWs/homework-2-main") # desktop

In [2]:
path = r"C:/Users/natal/OneDrive/Documents/GitHub/QMSS-GR5072-Spring2026-HWs/homework-2-main/Data"

In [3]:
os.getcwd()

'C:\\Users\\natal\\OneDrive\\Documents\\GitHub\\QMSS-GR5072-Spring2026-HWs\\homework-2-main'

In [4]:
# Load data in

nba_data = pd.read_csv(path + r"/Seasons_Stats_NBA.csv")

## Question #1: Which NBA player scored the most points in a single season? What seasons/year was this?

In [5]:
# sort by points (PTS, descending)
nba_data.head()

,Year,Player,Pos,Age,Tm,G,GS,MP,PER,TS%,...,FT%,ORB,DRB,TRB,AST,STL,BLK,TOV,PF,PTS
0,1950.0,Curly Armstrong,G-F,31.0,FTW,63.0,NaN,NaN,NaN,0.368,...,0.705,NaN,NaN,NaN,176.0,NaN,NaN,NaN,217.0,458.0
1,1950.0,Cliff Barker,SG,29.0,INO,49.0,NaN,NaN,NaN,0.435,...,0.708,NaN,NaN,NaN,109.0,NaN,NaN,NaN,99.0,279.0
2,1950.0,Leo Barnhorst,SF,25.0,CHS,67.0,NaN,NaN,NaN,0.394,...,0.698,NaN,NaN,NaN,140.0,NaN,NaN,NaN,192.0,438.0
3,1950.0,Ed Bartels,F,24.0,TOT,15.0,NaN,NaN,NaN,0.312,...,0.559,NaN,NaN,NaN,20.0,NaN,NaN,NaN,29.0,63.0
4,1950.0,Ed Bartels,F,24.0,DNN,13.0,NaN,NaN,NaN,0.308,...,0.548,NaN,NaN,NaN,20.0,NaN,NaN,NaN,27.0,59.0


In [6]:
nba_data_pts = nba_data.sort_values('PTS', ascending = False)
nba_data_pts

,Year,Player,Pos,Age,Tm,G,GS,MP,PER,TS%,...,FT%,ORB,DRB,TRB,AST,STL,BLK,TOV,PF,PTS
1706,1962.0,Wilt Chamberlain*,C,25.0,PHW,80.0,NaN,3882.0,31.7,0.536,...,0.613,NaN,NaN,2052.0,192.0,NaN,NaN,NaN,123.0,4029.0
1827,1963.0,Wilt Chamberlain*,C,26.0,SFW,80.0,NaN,3806.0,31.8,0.550,...,0.593,NaN,NaN,1946.0,275.0,NaN,NaN,NaN,136.0,3586.0
8469,1987.0,Michael Jordan*,SG,23.0,CHI,82.0,82.0,3281.0,29.8,0.562,...,0.857,166.0,264.0,430.0,377.0,236.0,125.0,272.0,237.0,3041.0
1593,1961.0,Wilt Chamberlain*,C,24.0,PHW,79.0,NaN,3773.0,27.8,0.519,...,0.504,NaN,NaN,2149.0,148.0,NaN,NaN,NaN,130.0,3033.0
1962,1964.0,Wilt Chamberlain*,C,27.0,SFW,80.0,NaN,3689.0,31.6,0.537,...,0.531,NaN,NaN,1787.0,403.0,NaN,NaN,NaN,182.0,2948.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
21678,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
22252,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
22864,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
23516,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


#### Wilt Chamberlain scored the most points (4029) in a single season (1962). 

## Question #2: Which NBA player scored the most points in 1991?

In [7]:
# sort by points (PTS, descending), filter for Year = 1991
nba_data_pts[nba_data_pts['Year'] == 1991]

,Year,Player,Pos,Age,Tm,G,GS,MP,PER,TS%,...,FT%,ORB,DRB,TRB,AST,STL,BLK,TOV,PF,PTS
10212,1991.0,Michael Jordan*,SG,27.0,CHI,82.0,82.0,3034.0,31.6,0.605,...,0.851,118.0,374.0,492.0,453.0,223.0,83.0,202.0,229.0,2580.0
10251,1991.0,Karl Malone*,PF,27.0,UTA,82.0,82.0,3302.0,24.8,0.596,...,0.770,236.0,731.0,967.0,270.0,89.0,79.0,244.0,268.0,2382.0
10132,1991.0,Patrick Ewing*,C,28.0,NYK,81.0,81.0,3104.0,23.7,0.561,...,0.745,194.0,711.0,905.0,244.0,80.0,258.0,291.0,287.0,2154.0
10280,1991.0,Chris Mullin*,SF,27.0,GSW,82.0,82.0,3315.0,21.4,0.618,...,0.884,141.0,302.0,443.0,329.0,173.0,63.0,245.0,176.0,2107.0
10414,1991.0,Dominique Wilkins*,SF,31.0,ATL,81.0,81.0,3078.0,23.5,0.555,...,0.829,261.0,471.0,732.0,265.0,123.0,65.0,201.0,156.0,2101.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
10224,1991.0,Bart Kofoed,SG,26.0,GSW,5.0,0.0,21.0,0.7,0.266,...,0.500,2.0,1.0,3.0,4.0,0.0,0.0,2.0,4.0,3.0
10385,1991.0,Jim Thomas,SG,30.0,MIN,3.0,0.0,14.0,0.7,0.250,...,NaN,0.0,0.0,0.0,1.0,1.0,0.0,1.0,0.0,2.0
10235,1991.0,Gary Leonard,C,23.0,ATL,4.0,0.0,9.0,8.7,0.568,...,0.500,0.0,2.0,2.0,0.0,0.0,1.0,0.0,2.0,2.0
10105,1991.0,Byron Dinkins,PG,23.0,IND,2.0,0.0,5.0,7.9,1.000,...,NaN,0.0,1.0,1.0,0.0,0.0,0.0,0.0,3.0,2.0


#### Michael Jordan scored the most points in 1991.

## Question #3: Which player had the best free throw percentage ("FT%") from the year 2000 to the most recent year in the data? If there are multiple players who have tied for the best percentage, then sort on the column with player names and submit your answer as a pd.DataFrame object.

In [8]:
# sort by FT% decreasing, filter for year = 2000+
nba_data_ft = nba_data.sort_values('FT%', ascending = False)
nba_data_ft

,Year,Player,Pos,Age,Tm,G,GS,MP,PER,TS%,...,FT%,ORB,DRB,TRB,AST,STL,BLK,TOV,PF,PTS
16099,2003.0,Bimbo Coles,PG,34.0,BOS,14.0,0.0,175.0,8.8,0.495,...,1.0,1.0,10.0,11.0,16.0,6.0,0.0,10.0,17.0,52.0
2165,1965.0,Bill McGill,C,25.0,LAL,8.0,NaN,37.0,10.3,0.367,...,1.0,NaN,NaN,12.0,3.0,NaN,NaN,NaN,6.0,15.0
16768,2004.0,James Jones,SF,23.0,IND,6.0,0.0,26.0,4.9,0.354,...,1.0,0.0,2.0,2.0,0.0,1.0,0.0,0.0,1.0,7.0
14562,2000.0,Pete Chilcutt,PF,31.0,UTA,26.0,0.0,224.0,5.7,0.374,...,1.0,15.0,28.0,43.0,10.0,5.0,4.0,9.0,31.0,47.0
14561,2000.0,Pete Chilcutt,PF,31.0,TOT,56.0,2.0,601.0,8.9,0.460,...,1.0,46.0,85.0,131.0,27.0,15.0,10.0,20.0,73.0,120.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
24568,2017.0,Damjan Rudez,SF,30.0,ORL,45.0,0.0,314.0,6.3,0.466,...,NaN,5.0,20.0,25.0,20.0,12.0,1.0,9.0,31.0,82.0
24620,2017.0,Mike Tobey,C,22.0,CHO,2.0,0.0,25.0,-0.1,0.250,...,NaN,2.0,1.0,3.0,1.0,0.0,0.0,1.0,2.0,2.0
24622,2017.0,Axel Toupane,SF,24.0,TOT,4.0,0.0,47.0,6.2,0.611,...,NaN,0.0,1.0,1.0,0.0,1.0,1.0,0.0,6.0,11.0
24623,2017.0,Axel Toupane,SF,24.0,MIL,2.0,0.0,6.0,-9.9,0.000,...,NaN,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0


In [9]:
nba_data2000 = nba_data_ft[nba_data_ft['Year'] >= 2000]
nba_data2000

,Year,Player,Pos,Age,Tm,G,GS,MP,PER,TS%,...,FT%,ORB,DRB,TRB,AST,STL,BLK,TOV,PF,PTS
16099,2003.0,Bimbo Coles,PG,34.0,BOS,14.0,0.0,175.0,8.8,0.495,...,1.0,1.0,10.0,11.0,16.0,6.0,0.0,10.0,17.0,52.0
16768,2004.0,James Jones,SF,23.0,IND,6.0,0.0,26.0,4.9,0.354,...,1.0,0.0,2.0,2.0,0.0,1.0,0.0,0.0,1.0,7.0
14562,2000.0,Pete Chilcutt,PF,31.0,UTA,26.0,0.0,224.0,5.7,0.374,...,1.0,15.0,28.0,43.0,10.0,5.0,4.0,9.0,31.0,47.0
14561,2000.0,Pete Chilcutt,PF,31.0,TOT,56.0,2.0,601.0,8.9,0.460,...,1.0,46.0,85.0,131.0,27.0,15.0,10.0,20.0,73.0,120.0
18893,2008.0,Ronald Dupree,SF,27.0,SEA,4.0,0.0,18.0,17.5,0.515,...,1.0,1.0,7.0,8.0,1.0,1.0,0.0,0.0,0.0,4.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
24568,2017.0,Damjan Rudez,SF,30.0,ORL,45.0,0.0,314.0,6.3,0.466,...,NaN,5.0,20.0,25.0,20.0,12.0,1.0,9.0,31.0,82.0
24620,2017.0,Mike Tobey,C,22.0,CHO,2.0,0.0,25.0,-0.1,0.250,...,NaN,2.0,1.0,3.0,1.0,0.0,0.0,1.0,2.0,2.0
24622,2017.0,Axel Toupane,SF,24.0,TOT,4.0,0.0,47.0,6.2,0.611,...,NaN,0.0,1.0,1.0,0.0,1.0,1.0,0.0,6.0,11.0
24623,2017.0,Axel Toupane,SF,24.0,MIL,2.0,0.0,6.0,-9.9,0.000,...,NaN,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0


In [10]:
# multiple people have 1.0 FT%, make into df
nba_data_df = pd.DataFrame(nba_data2000[nba_data2000['FT%'] == 1.0])

#### I'm not sure where sorting by player name came in, but I was able to make a dataframe by filtering for FT% = 1.0, which was the highest FT%.

## Question #4: Among those with at least 50 attempts ('FTA'), which player had the best free throw percentage ("FT%") from the year 2000 to the most recent year in the data?

In [11]:
# filter for FTA >= 50 & year >= 2000
nba_data_fta = nba_data[(nba_data_ft['FTA'] >= 50) & (nba_data['Year'] >= 2000)]

# sort by FT%
nba_data_fta = nba_data_fta.sort_values('FT%', ascending = False)
nba_data_fta

,Year,Player,Pos,Age,Tm,G,GS,MP,PER,TS%,...,FT%,ORB,DRB,TRB,AST,STL,BLK,TOV,PF,PTS
15565,2002.0,Terrell Brandon,PG,31.0,MIN,32.0,28.0,962.0,21.2,0.494,...,0.988,17.0,76.0,93.0,264.0,52.0,6.0,43.0,49.0,397.0
19434,2009.0,Jose Calderon,PG,27.0,TOR,68.0,68.0,2333.0,18.8,0.613,...,0.981,16.0,178.0,194.0,607.0,74.0,7.0,143.0,114.0,873.0
22484,2014.0,Danny Granger,SF,30.0,IND,29.0,2.0,653.0,10.3,0.491,...,0.962,24.0,81.0,105.0,33.0,9.0,13.0,36.0,41.0,242.0
22671,2014.0,Gary Neal,SG,29.0,CHA,22.0,1.0,507.0,14.0,0.571,...,0.961,2.0,38.0,40.0,37.0,11.0,0.0,25.0,39.0,247.0
19350,2009.0,Ray Allen,SG,33.0,BOS,79.0,79.0,2876.0,17.3,0.624,...,0.952,66.0,212.0,278.0,218.0,69.0,13.0,134.0,157.0,1434.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
14837,2000.0,Olden Polynice,C,35.0,UTA,82.0,79.0,1819.0,10.4,0.497,...,0.311,166.0,287.0,453.0,37.0,30.0,84.0,70.0,260.0,435.0
17715,2006.0,Andris Biedrins,C,19.0,GSW,68.0,2.0,1000.0,13.6,0.595,...,0.306,124.0,159.0,283.0,24.0,23.0,47.0,34.0,190.0,258.0
23225,2015.0,Ian Mahinmi,C,28.0,IND,61.0,6.0,1146.0,10.8,0.516,...,0.304,105.0,251.0,356.0,33.0,30.0,46.0,59.0,172.0,265.0
23026,2015.0,Joey Dorsey,PF,31.0,HOU,69.0,17.0,854.0,11.1,0.507,...,0.289,103.0,176.0,279.0,25.0,40.0,25.0,41.0,157.0,184.0


#### Terrell Brandon in 2002 had the best free throw percentage (98%) of all players who had at least 50 attempts.

## Question #5: Rename the variable "Pos" to "Position".

In [12]:
nba_data.rename(columns={'Pos':'Position'}, inplace=True)

## Question #6: Use this variable to create two variables that are called "first_position" and "second_position". Hint: If a player plays two positions, then the "-" is used as a delimiter between their first and second positions. In these cases, you will have to split the position variable into two.

In [41]:
nba_data[['first_position', 'second_position']] = nba_data['Position'].str.split('-', n =1, expand = True)
# used code from https://stackoverflow.com/questions/14745022/how-to-split-a-dataframe-string-column-into-two-columns

nba_data['second_position'].replace(to_replace=[None], value=np.nan, inplace=True)
# https://stackoverflow.com/questions/23743460/replace-none-with-nan-in-pandas-dataframe

nba_data.head() # worked!

C:\Users\natal\AppData\Local\Temp\ipykernel_20392\2554002844.py:4: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  nba_data['second_position'].replace(to_replace=[None], value=np.nan, inplace=True)


,Year,Player,Position,Age,Tm,G,GS,MP,PER,TS%,...,TRB,AST,STL,BLK,TOV,PF,PTS,first_position,second_position,position_united
0,1950.0,Curly Armstrong,G-F,31.0,FTW,63.0,NaN,NaN,NaN,0.368,...,NaN,176.0,NaN,NaN,NaN,217.0,458.0,G,F,G
1,1950.0,Cliff Barker,SG,29.0,INO,49.0,NaN,NaN,NaN,0.435,...,NaN,109.0,NaN,NaN,NaN,99.0,279.0,SG,NaN,SG
2,1950.0,Leo Barnhorst,SF,25.0,CHS,67.0,NaN,NaN,NaN,0.394,...,NaN,140.0,NaN,NaN,NaN,192.0,438.0,SF,NaN,SF
3,1950.0,Ed Bartels,F,24.0,TOT,15.0,NaN,NaN,NaN,0.312,...,NaN,20.0,NaN,NaN,NaN,29.0,63.0,F,NaN,F
4,1950.0,Ed Bartels,F,24.0,DNN,13.0,NaN,NaN,NaN,0.308,...,NaN,20.0,NaN,NaN,NaN,27.0,59.0,F,NaN,F


## Question #7: Combine these two variables back into a single variable called "position_united". Make sure to separate the two positions by "-", as the original data field did. If there is no second position, then there will only be a single position. Again, this is similar to how the original variable was coded. Present a frequency table of "position_united".

In [44]:
nba_data['position_united'] = nba_data["first_position"] + '-' + nba_data["second_position"]

nba_data.head()

,Year,Player,Position,Age,Tm,G,GS,MP,PER,TS%,...,TRB,AST,STL,BLK,TOV,PF,PTS,first_position,second_position,position_united
0,1950.0,Curly Armstrong,G-F,31.0,FTW,63.0,NaN,NaN,NaN,0.368,...,NaN,176.0,NaN,NaN,NaN,217.0,458.0,G,F,G-F
1,1950.0,Cliff Barker,SG,29.0,INO,49.0,NaN,NaN,NaN,0.435,...,NaN,109.0,NaN,NaN,NaN,99.0,279.0,SG,NaN,NaN
2,1950.0,Leo Barnhorst,SF,25.0,CHS,67.0,NaN,NaN,NaN,0.394,...,NaN,140.0,NaN,NaN,NaN,192.0,438.0,SF,NaN,NaN
3,1950.0,Ed Bartels,F,24.0,TOT,15.0,NaN,NaN,NaN,0.312,...,NaN,20.0,NaN,NaN,NaN,29.0,63.0,F,NaN,NaN
4,1950.0,Ed Bartels,F,24.0,DNN,13.0,NaN,NaN,NaN,0.308,...,NaN,20.0,NaN,NaN,NaN,27.0,59.0,F,NaN,NaN


In [47]:
nba_data['position_united'].value_counts() # from Prof. Hicks' code

position_united
F-C      81
G-F      79
F-G      60
C-F      35
C-PF     33
SF-SG    30
PG-SG    28
PF-C     27
SG-PG    27
SG-SF    26
PF-SF    25
SF-PF    21
SG-PF     3
C-SF      2
SF-PG     1
PG-SF     1
Name: count, dtype: int64

##  Question #8: Create two new datasets.
### a new dataset from the original dataset that includes all data except the age variable (be sure to give this dataset a new name).
### a new dataset from the original dataset that includes the year, the player name, and age.
### add a new column to both datasets called mergeid that includes a sequence of numbers beginning with a 1 in the first row of the data and ending with the total number of rows in the last row of the data

In [15]:
# a new dataset from the original dataset that includes all data except the age variable (be sure to give this dataset a new name).
nba_data_noage = nba_data.drop('Age', axis = 1)

In [16]:
### a new dataset from the original dataset that includes the year, the player name, and age.
nba_data_3 = nba_data.loc[:, ['Player', 'Age', 'Year']]

In [28]:
### add a new column to both datasets called mergeid that includes a sequence of numbers beginning with a 1 in the first row of the data and ending with the total number of rows in the last row of the data
nba_data_noage['mergeid'] = np.arange(0,24691,1)
nba_data_3['mergeid'] = np.arange(0,24691,1)
# from Prof Hicks' code

,Player,Age,Year,mergeid
0,Curly Armstrong,31.0,1950.0,0
1,Cliff Barker,29.0,1950.0,1
2,Leo Barnhorst,25.0,1950.0,2
3,Ed Bartels,24.0,1950.0,3
4,Ed Bartels,24.0,1950.0,4


## Question #9: Join the two datasets from question (8) together to recreate the original dataset plus the new merge id.

In [31]:
full_data = pd.merge(nba_data_3, nba_data_noage, on=["mergeid", "Player", "Year"])
full_data.head()
# code from Prof. Hicks'

,Player,Age,Year,mergeid,Position,Tm,G,GS,MP,PER,...,TRB,AST,STL,BLK,TOV,PF,PTS,first_position,second_position,position_united
0,Curly Armstrong,31.0,1950.0,0,G-F,FTW,63.0,NaN,NaN,NaN,...,NaN,176.0,NaN,NaN,NaN,217.0,458.0,G,F,G-F
1,Cliff Barker,29.0,1950.0,1,SG,INO,49.0,NaN,NaN,NaN,...,NaN,109.0,NaN,NaN,NaN,99.0,279.0,SG,,SG-
2,Leo Barnhorst,25.0,1950.0,2,SF,CHS,67.0,NaN,NaN,NaN,...,NaN,140.0,NaN,NaN,NaN,192.0,438.0,SF,,SF-
3,Ed Bartels,24.0,1950.0,3,F,TOT,15.0,NaN,NaN,NaN,...,NaN,20.0,NaN,NaN,NaN,29.0,63.0,F,,F-
4,Ed Bartels,24.0,1950.0,4,F,DNN,13.0,NaN,NaN,NaN,...,NaN,20.0,NaN,NaN,NaN,27.0,59.0,F,,F-


## Question #10: Subset the original dataset to the years 1995-1997. Group the data by team name and year and then summarize the average number of points per team. Arrange from most to least points. (Hint: Your answer should have a shape of 88 x 3, since we are looking at the average points scored over the 1995-1997 seasons. Note that 2 of these columns could be indexes, in which case Pandas will tell you your data is 88 rows x 1 columns. Finally, also note that the data for VAN and TOR are missing in 1995 as these teams did not exist in that year. So your answer could also be 90 x 3, if you include missing/empty data in these rows).

In [19]:
nba_data_filter = nba_data[(nba_data['Year'] >= 1995) & (nba_data['Year'] <= 1997)].reset_index()
# filtering for years = 1995-1997

In [20]:
nba_data_filter = nba_data_filter.groupby(['Year', 'Tm']).mean('PTS').reset_index() 
#grouping by Year and Team and averaging points

nba_data_filter = nba_data_filter.loc[:, ['Year', 'Tm', 'PTS']] 
# had to slice for some reason to get the correct number of columns

In [21]:
nba_data_filter = nba_data_filter.sort_values('PTS', ascending = False)
# sorting by points

nba_data_filter.head()

## Question #11: Similar to Q10, start by subsetting the original dataset to the years 1995-1997. Then create a wide dataset that keeps each year in its own row, but spreads team names to multiple columns with each column delineating points per team in each of the 1995-1997 seasons (Hint: Your final answer should have a shape of 3 x 30).

In [22]:
# filtering for 1995-1997
nba_data_filter2 = nba_data[(nba_data['Year'] >= 1995) & (nba_data['Year'] <= 1997)].reset_index()

In [23]:
# grouping by Year and Team, averaging points
nba_data_filter2 = nba_data_filter2.groupby(['Year', 'Tm']).mean('PTS').reset_index()

In [24]:
# moving to wide format
pivoted = nba_data_filter2.pivot(index = 'Year', columns='Tm', values='PTS')
# code from Prof. Hicks'

In [25]:
pivoted.head()

Tm,ATL,BOS,CHH,CHI,CLE,DAL,DEN,DET,GSW,HOU,...,PHO,POR,SAC,SAS,SEA,TOR,TOT,UTA,VAN,WSB
Year,,,,,,,,,,,,,,,,,,,,,
1995.0,440.055556,495.764706,515.562500,520.312500,494.466667,604.428571,488.764706,503.312500,481.500000,499.470588,...,604.866667,528.187500,537.066667,546.375000,646.785714,NaN,488.681818,581.733333,NaN,515.125000
1996.0,447.722222,471.944444,401.476190,575.000000,498.200000,560.600000,471.352941,558.714286,490.235294,466.888889,...,503.058824,543.000000,510.187500,565.133333,659.384615,420.736842,419.767857,600.285714,409.0000,400.380952
1997.0,457.294118,458.222222,450.444444,563.866667,551.769231,275.222222,348.695652,482.687500,510.687500,458.222222,...,366.565217,427.052632,494.250000,370.900000,517.125000,372.809524,322.380952,563.600000,457.0625,543.133333


## Question #12: Now return the data from #11 back into a long (tidy) format by moving Team (TM), Points and Year into their own columns (Hint: Your data shape should be 90 x 3).

In [26]:
# moving back to long format
long_data = (pivoted.stack()   
              .reset_index()
              .rename(columns={0: "PTS"}))
# code from Prof. Hicks'

In [ ]:
long_data.head()

# Bibliography
- Leo Rochael. (2016, September 6). *Answer to: How to split a dataframe string column into two columns?* \[Source code\]. Stack Overflow. https://stackoverflow.com/questions/14745022/how-to-split-a-dataframe-string-column-into-two-columns
- Nickolai. (2018, September 24). *Answer to: Replace None with NaN in pandas dataframe* \[Source code\]. https://stackoverflow.com/questions/23743460/replace-none-with-nan-in-pandas-dataframe
- Guillaume Jacquenot. (2014, May 19). *Answer to: Replace None with NaN in pandas dataframe* \[Source code\]. https://stackoverflow.com/questions/23743460/replace-none-with-nan-in-pandas-dataframe
- GeeksforGeeks. (2025, July 11). *Creating a Pandas Series* \[Source code\]. https://www.geeksforgeeks.org/pandas/creating-a-pandas-series/
- Hicks, R. (2026, January 30). *week_02.ipynb* \[Source code\]. Jupyter Notebook
- Hicks, R. (2026, February 5). *week_03.ipynb* \[Source code\]. Jupyter Notebook
- Hicks, R. (2026, January 17). *week_04.ipynb* \[Source code\]. Jupyter Notebook